### PR TITLE
Prepare plugin for optional direct inclusion into Zeek.

### PR DESCRIPTION
**Pull request by bbannier**

### CODE REVIEW

**Review by rsmmr — APPROVED** [review]
I assume this means that we will inject everything that `configure`  would normally set from Zeek's cmake?

**Comment by bbannier** [comment]
> I assume this means that we will inject everything that `configure` would normally set from Zeek's cmake?

Yes, exactly. Currently this seems sufficient: https://github.com/zeek/zeek/blob/8d28cfa320143d603ac8facc42c3ac8d2b99c638/CMakeLists.txt#L371-L384.

**Comment by bbannier** [comment]
Thanks for the review @rsmmr. I'll put this back into draft mode and will keep pushing changes needed for Zeek integration (zeek/zeek#2006) to this branch.

File: cmake/FindSpicy.cmake [hunk 1]
```diff
@@ -11,17 +11,10 @@
 #
 #     If SPICY_FOUND is true:
 #
-#         SPICYC                        full path to spicyc
```
Comment (rsmmr) [inline cmake/FindSpicy.cmake:14]: I'm just realizing something: analyzers are pulling this in indirectly through `find_package(Spicy)` in `FindSpicyPlugin.cmake`. Could they be using any of the removed values?

File: cmake/FindSpicy.cmake [hunk 2]
```diff
@@ -141,12 +130,8 @@ function (spicy_print_summary)
             "\nFound Spicy:           ${HAVE_SPICY}")
 
     if (HAVE_SPICY)
-        message(
-            "\nVersion:               ${SPICY_VERSION} (${SPICY_VERSION_NUMBER})"
-            "\nPrefix:                ${SPICY_PREFIX}"
```
Comment (rsmmr) [inline cmake/FindSpicy.cmake:146]: This one would actually be useful to keep for debugging what Spicy installation is being used. We could just set it to some descriptive string when building from inside Zeek.

File: cmake/FindSpicy.cmake (hunk 1 above)
Comment (bbannier) [reply cmake/FindSpicy.cmake:14]: Yes, they might be. Should I drop the cleanup patch?

Comment (rsmmr) [reply cmake/FindSpicy.cmake:14]: Well, yes, but when building from Zeek, we then actually need to set them to appropriate values too.

File: cmake/FindSpicy.cmake (hunk 2 above)
Comment (bbannier) [reply cmake/FindSpicy.cmake]: I will keep this part as is.

**Review by rsmmr — APPROVED** [review]

**Pull request merged**